File: Python/July_problems/problem1823_8th_july_2024.py
def findTheWinner(n: int, k: int) -> int:
    player = [i for i in range(1, n+1)]
    current = 0
    while len(player) != 1:
        current = (current + (k-1)) % len(player)
        player.pop(current)
    return player[0]


def problem1823():
    print(findTheWinner(5, 2))


if __name__ == "__main__":
    problem1823()

File: CPP/Year-2024/July_Problems/problem1823_8th_july_2024.cpp
#include <iostream>
#include <vector>

using namespace std;

int findTheWinner(int n, int k)
{
    vector<int> player;
    for (int i = 1; i <= n; i += 1)
    {
        player.push_back(i);
    }
    int current = 0;
    while (player.size() != 1)
    {
        current = (current + (k - 1)) % player.size();
        player.erase(player.begin() + current);
    }
    return player[0];
}

int main()
{
    cout << findTheWinner(5, 2);
    return 0;
}